### ISSUE TITLE
Support for new android Observable* types

### ISSUE THREAD

**Issue by dozd** [johncarl81/parceler#226]
Hello, It would be great to provide native support also for new Observable* types used in Android data binding. As they are mainly being used in models that should be parcelable (to save state) it will be common case.

What do you think?

**Comment by dozd** [comment]
I mean these types can be used now (as they are Serializable). But if i have custom type `ObservableField<CustomType>` i want `CustomType` annotate with `Parcel`. Now it is parcelled by serialization but it would be better to unwrap it from Observable and parcell it through library like `CystomType` directly.

**Comment by johncarl81** [comment]
Sounds like a great idea.  All we have to do is add a matcher here: https://github.com/johncarl81/parceler/blob/master/parceler/src/main/java/org/parceler/internal/ParcelerModule.java#L221  like so: 
```java
Matchers.type(new ASTStringType("android.databinding.ObservableField")).ignoreGenerics().build()
```
And a custom `ReadWriteGenerator` to handle the `get()` and `set()` calls.

**Comment by johncarl81** [comment]
Or maybe we could do better with the matcher and also determine if the given generic is configured as a `@Parcel`.

**Comment by sevar83** [comment]
Can't we just use `ObservableParcelable` instead of `ObservableField` !?

**Comment by johncarl81** [comment]
Yes, of course.  Parceler should recognize the `Pacelable` inheritance and marshal accordingly.  The only problem there is that your target object needs to implement `Parcelable` as well.

**Comment by johncarl81** [comment]
Oh, and @dozd, for the time being you could always write your own `ParcelConverter` to handle the serialization of `ObservableField`.

**Comment by johncarl81** [comment]
@dozd @sevar83 this is implemented on the referenced PR.  Care to try it out?

**Comment by dozd** [comment]
@johncarl81 I could. Is ObservableArrayList also supported? :) That was my case in real app.

**Comment by johncarl81** [comment]
Not currently.  Are you looking to use this directly with the Parcels utility class btw?:
```
ObservableArrayList list = new ObservableArrayList();
Parcels.wrap(list);
```

**Comment by dozd** [comment]
Just like you parcel Array of any of the mapped types, you can parcel ObservableArrayList.

```
@Parcel
class A {
  ObservableField<B> b;
  ObservableArrayList<B> listB;
}

@Parcel
class B {
}

a = new A();
Parcels.wrap(a)

**Comment by johncarl81** [comment]
@dozd, ok, that PR shoudl now support `ObservableArrayList` (and `ObservableArrayMap`).  Let me know if it works for you.

**Comment by dozd** [comment]
Hi @johncarl81, `ObservableArrayList` is working for me. Thanks!

**Comment by dozd** [comment]
@johncarl81 As well as `ObservableField` that I've tried today. You can merge it :+1: .

**Comment by johncarl81** [comment]
Great @dozd, thanks for testing.

FYI, I think I am going to pass on native wrapping / unwrapping of native `ObservableArrayList` and `observableField` to avoid the dependency on Android.  So this scenario will not work:
```
ObservableArrayList list = new ObservableArrayList();//...
Parcels.wrap(list);
```

**Comment by dozd** [comment]
I think, it's ok.

**Comment by johncarl81** [comment]
Ok, this will be released in 1.1.6
